controllers/secret_controller.go

```diff
@@ -147,7 +151,9 @@ func (s *SecretSyncController) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	// Update the selector in cache always (comparing is pointless)
+	s.selectorMutex.Lock()
```

Comment:
I feel like we might be well served by defining a custom type for the selector cache and defining methods for accessing it. That would keep the concurrency aspect encapsulated and ensure reads and writes are done in a consistent fashion. wdyt?

Reply:
I think it would hide a pretty simple implementation unnecessarily.